api/channel.go

```diff
@@ -397,10 +397,15 @@ func updateChannelPurpose(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getChannels(c *Context, w http.ResponseWriter, r *http.Request) {
+	// check if the team exist
+	if result := <-Srv.Store.Team().Get(c.TeamId); result.Err != nil {
```

Comment:
Yes please, can you follow the pattern of the error checking here https://github.com/mattermost/platform/blob/master/api/context.go#L220 and then add a function that checks if `c.Session.GetTeamByTeamId(teamId) == nil` then error out by setting the context error. That should work, if not let me know and I'll help you debug